src/Analyzers/CSharp/Analyzers/AddRequiredParentheses/CSharpAddRequiredPatternParenthesesDiagnosticAnalyzer.cs

```diff
@@ -29,11 +29,11 @@ public CSharpAddRequiredPatternParenthesesDiagnosticAnalyzer()
         protected override ImmutableArray<SyntaxKind> GetSyntaxNodeKinds()
             => s_kinds;
 
-        protected override int GetPrecedence(BinaryPatternSyntax pattern)
+        protected override int GetPrecedence(PatternSyntax pattern)
             => (int)pattern.GetOperatorPrecedence();
 
         protected override bool IsBinaryLike(PatternSyntax node)
-            => node is BinaryPatternSyntax;
+            => node is BinaryPatternSyntax or SlicePatternSyntax;
```

Comment:
this is an actual question on my part.  why is SlicePattern *not* just a BinaryPattern?  is it because it last no left hand side?

Reply:
Correct. Except for this trick to re-use existing paths in AddRequiredParens it would not have occured to me to make the SlicePattern be a BinaryPattern.
```
slice_pattern
  : '..' pattern?
  ;
  ```

Reply:
if that's hte gramamr for SlicePAttern, then thne SyntaxModel is correct IMO and this code change is appropriate.  Thanks!